specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-03-01-preview/LongTermRetention.json

```diff
@@ -0,0 +1,411 @@
+            "schema": {
+              "$ref": "#/definitions/LtrPreBackupRequest"
+            }
+          },
+          {
+            "name": "x-ms-client-request-id",
+            "in": "header",
+            "description": "A unique ID provided by the client to identify the specific request.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok. API finished synchronously with Success.",
+            "schema": {
+              "$ref": "#/definitions/LtrPreBackupResponse"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error Scenario 4xx/5xx",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "ErrorCode string in the event of a failure.",
+                "type": "string"
+              },
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Sample_Prebackup": {
+            "$ref": "./examples/LongTermRetentionPreBackup.json"
+          }
+        },
+        "deprecated": false
+      }
+    }
+  },
+  "definitions": {
+    "BackupRequestBase": {
+      "description": "BackupRequestBase is the base for all backup request.",
+      "required": [
+        "backupSettings"
+      ],
+      "type": "object",
+      "properties": {
+        "backupSettings": {
+          "$ref": "#/definitions/BackupSettings",
+          "description": "Backup Settings"
+        }
+      }
+    },
+    "LtrBackupRequest": {
+      "description": "The request that is made for a backup.",
+      "type": "object",
+      "properties": {
+        "targetDetails": {
+          "$ref": "#/definitions/BackupStoreDetails",
+          "description": "Backup store detail for target server"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRequestBase"
+        }
+      ],
+      "required": [
+        "targetDetails"
+      ]
+    },
+    "LtrPreBackupRequest": {
+      "description": "A request that is made for pre-backup.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRequestBase"
+        }
+      ]
+    },
```

Comment:
Curious on the reason for making this a separate type:
Shouldn't it just always use the exact same request as ltrBackupRequest, if its going to set up / validate the exact same work? Do you foresee a future where they diverge?

Reply:
As confirmed with Azure Data backup team, the SAUri list is never required for performing validation. Hence we have separate types created for validation and actual backup request